pkg/gds/models/v1/models.go

```diff
@@ -173,6 +176,69 @@ func AppendCertReqID(vasp *pb.VASP, certreqID string) (err error) {
 	return nil
 }
 
+// NewCertificateRequest creates and returns a new certificate request to be associated with a VASP.
+func NewCertificateRequest(vasp *pb.VASP) (certRequest *CertificateRequest, err error) {
+	var (
+		organizationName    string
+		localityName        string
+		stateOrProvinceName string
+		countryName         string
+	)
+
+	if vasp == nil {
+		return nil, errors.New("must supply a VASP object for certificate request creation")
+	}
+
+	certRequest = &CertificateRequest{
+		Id:         uuid.New().String(),
+		Vasp:       vasp.Id,
+		CommonName: vasp.CommonName,
+		Params:     make(map[string]string),
+	}
+
+	// Populate the organization name, if available.
+	if vasp.Entity.Name != nil {
+		for _, name := range vasp.Entity.Name.NameIdentifiers {
+			if name.LegalPersonNameIdentifierType == ivms101.LegalPersonLegal {
+				organizationName = name.LegalPersonName
+				break
+			}
+		}
+	}
+	if organizationName != "" {
+		certRequest.Params["organizationName"] = organizationName
+	} else {
+		log.Info().
+			Str("vasp_id", vasp.Id).
+			Str("certreq_id", certRequest.Id).
+			Msg("organization name not found, populating new certificate request with default value")
+		certRequest.Params["organizationName"] = "TRISA Production"
+	}
+
+	// Populate the location information, if available.
+	if len(vasp.Entity.GeographicAddresses) > 0 {
+		address := vasp.Entity.GeographicAddresses[0]
+		localityName = address.TownLocationName
+		stateOrProvinceName = address.CountrySubDivision
+		countryName = address.Country
+	}
+	if localityName != "" && stateOrProvinceName != "" && countryName != "" {
+		certRequest.Params["localityName"] = localityName
+		certRequest.Params["stateOrProvinceName"] = stateOrProvinceName
+		certRequest.Params["countryName"] = countryName
+	} else {
+		log.Info().
```

Comment:
Could we change this to `Debug` - it will likely be the common case for a while.